### ISSUE TITLE
Configuration Option for Default Job Working Directory

### ISSUE THREAD

**Issue by jmiahjones** [rstudio/rstudio#8985]
The default behavior for creating a new job is to set the working directory to the parent of the currently selected script.
This is a bit problematic for me for a number of reasons:
1. The file path box is not very large, so if I open any path that is not under ~/, I can't really see what the default working directory is. E.g. when mounting a secondary storage drive on Ubuntu and working from there.
![image](https://user-images.githubusercontent.com/22567790/108231902-3b1b6a80-7110-11eb-8fb8-3bb7ed04b503.png)
Although the script file is also affected, it is generally more intuitive to select the script to me.
2. I can neither edit the text in the working path box, nor can I obtain focus and scroll to the right if the path happens to overrun the box. 
3. The working directory might change depending on which script I happen to have open in the source editor.

While the editing problems have been mentioned before https://github.com/rstudio/rstudio/issues/5811, editing the path is not necessarily the issue. This issue would be mitigated if I could enforce the "current project working directory" that is the [default and recommended behavior](https://support.rstudio.com/hc/en-us/articles/200711843-Working-Directories-and-Workspaces). 

Others seem to have issues with this as well: 

- https://community.rstudio.com/t/default-working-directory-of-jobs-within-a-project/22178 
- https://community.rstudio.com/t/default-working-directory-and-output-options-when-launching-jobs-in-project/70345
- https://github.com/rstudio/rstudio/issues/6036

The Terminal Shell has a configuration option to set the initial directory as either the Project, Current, or Home directory.  Ostensibly, a similar config option could be created for the Jobs Launcher.
![Screenshot from 2021-02-17 10-55-02](https://user-images.githubusercontent.com/22567790/108230519-e0cdda00-710e-11eb-842e-8931748e50e8.png).

Alternative options I can think of:
- Default to the current working directory.
- Use the _existing_ terminal default path configuration option, although some of the labeling around what this affects would have to change.
- Making the text boxes editable, as requested in #6036, although  @gtritchie has mentioned in #5811 how this creates other problems.

The defaulting behavior seems to be hard-coded [here](https://github.com/rstudio/rstudio/blob/b3928c76bad79acc6e4eff7cfc1520b3c7b89224/src/gwt/src/org/rstudio/studio/client/workbench/views/jobs/view/JobLauncherDialog.java#L39).

**Comment by ronblum** [comment]
@jmiahjones Thanks for raising this and the suggestions! We'll review this as part of our ongoing development of RStudio. In the meantime, for the second half of part (2), "nor can I obtain focus and scroll to the right if the path happens to overrun the box," if you click in the box and hold it and move left and right you can see the full path. Though I realize this isn't what you're looking for, it's just one piece.

**Comment by jmiahjones** [comment]
@ronblum Sounds good!

For the second part,  this might be a bug. On my Ubuntu 20.04.1 system, if I drag the cursor to the right, it does not move focus. The arrow keys also do nothing. Should I log a separate bug as well?

Below is me dragging the selection text all the way to the right.
![image](https://user-images.githubusercontent.com/22567790/108525442-12c17680-729e-11eb-8bfc-0256ba5bbe85.png)

**Comment by ronblum** [comment]
@jmiahjones Yes, that would be great, thanks! I imagine that if it can be done on a Mac it ought to be possible on Ubuntu, though there might be OS limitations on that.

**Comment by AltfunsMA** [comment]
I came here to post a similar enhancement request. My preference would be for the first two options that @jmiahjones mentions to make the default behaviour configurable.

I have the job launcher on a keyboard shortcut but often my scripts are written with relative paths from the project directory, so I can't simply hit enter.

**Comment by mpaulacaldas** [comment]
I would like to add my vote for this enhancement request 🙋 The current default doesn't work well with the project-oriented workflows that a large part of the R user base has adopted, in part thanks to many of the educational resources created by RStudio (e.g. R4DS, rstats.wtf)

**Comment by kevinushey** [comment]
As a workaround until this changes in RStudio, I think using [here](https://here.r-lib.org/) may be useful.

**Comment by nickreich** [comment]
Adding a +1 for this enhancement request. The ability to not have to manually choose the directory of the script every time you want to run a local job would be awesome. thanks!

**Comment by ValValetl** [comment]
Also +1 for this request. perhaps it would be nice if this could be set via a environment variable?

**Comment by ronblum** [comment]
@ValValetl Can you also provide a 👍  in the original post? It's one way in which we prioritize issues. I can't guarantee if and when this would be changed, but it helps decide what to include.

(@nickreich I see that you already did so. We check, we really check! 🙂 )

**Comment by ValValetl** [comment]
sure, did not know that this is how it works ;)

**Comment by ronblum** [comment]
@ValValetl 👍
